### ISSUE TITLE
Assertion "Wrong MachineOperand accessor" failed (from SPIRVOpenCLBIFs.cpp)

### ISSUE THREAD

**Issue by iliya-diyachkov** [KhronosGroup/LLVM-SPIRV-Backend#9]
There are 3 tests (transcoding/OpImageSampleExplicitLod.ll, transcoding/OpImageReadMS.ll, SamplerArgNonKernel.ll) with the fail:
const llvm::ConstantInt* llvm::MachineOperand::getCImm() const: Assertion `isCImm() && "Wrong MachineOperand accessor"' failed.

All three come from SPIRVOpenCLBIFs.cpp so they may be fixed in new OpenCl builtins implementation:
```
#10 0x000056284e9e1959 llvm::MachineOperand::getCImm() const llvm/include/llvm/CodeGen/MachineOperand.h:541:5
#11 0x000056284e9eeb79 getLiteralValueForConstant(llvm::Register, llvm::MachineRegisterInfo const*) llvm/lib/Target/SPIRV/SPIRVOpenCLBIFs.cpp:233:59
#12 0x000056284e9ef855 genSampledReadImage(llvm::MachineIRBuilder&, llvm::Register, llvm::MachineInstr const*, llvm::SmallVectorImpl<llvm::Register> const&, llvm::SPIRVTypeRegistry*) llvm/lib/Target/SPIRV/SPIRVOpenCLBIFs.cpp:424:47
```

**Comment by iliya-diyachkov** [comment]
constVReg in getLiteralValueForConstant is expected to be initialized by a constant. However in these tests it has different sources:
transcoding/OpImageSampleExplicitLod.ll (and similar in SamplerArgNonKernel.ll):
``
  %4:_(s32) = OpFunctionParameter %3:type(s32)
``
transcoding/OpImageReadMS.ll
``
%31:_(<2 x s32>) = G_INSERT_VECTOR_ELT %29:_, %17:_(s32), %18:_(s32)
``
I think these cases should be supported when new implementation of builtins is completed.

**Comment by iliya-diyachkov** [comment]
The issue was fixed in 50224ca241b606d4aa22528b1ac81c33acb1b8ef . transcoding/OpImageSampleExplicitLod.ll, transcoding/OpImageReadMS.ll are passed, SamplerArgNonKernel.ll fails with different issue.
